drivers/display/display_st7789h2.c

```diff
@@ -0,0 +1,608 @@
+			     const enum display_pixel_format pixel_format)
+{
+
+	if (pixel_format == PIXEL_FORMAT_RGB_565) {
+		return 0;
+	}
+	LOG_ERR("Pixel format change not implemented");
+	return -ENOTSUP;
+}
+
+static int st7789h2_set_orientation(const struct device *dev,
+			    const enum display_orientation orientation)
+{
+	if (orientation == DISPLAY_ORIENTATION_NORMAL) {
+		return 0;
+	}
+	LOG_ERR("Changing display orientation not implemented");
+	return -ENOTSUP;
+}
+
+static void st7789h2_lcd_init(const struct device *dev)
+{
+	uint8_t tmp;
+
+	struct st7789h2_data *data = (struct st7789h2_data *)dev->data;
+	const struct st7789h2_config *config = (const struct st7789h2_config *)dev->config;
+
+	st7789h2_set_lcd_margins(data, data->x_offset,
+				data->y_offset);
+
+	st7789h2_transmit(dev, ST7789H2_CMD_CMD2EN, (uint8_t *) config->cmd2en_param,
+			 sizeof(config->cmd2en_param));
+
+	st7789h2_transmit(dev, ST7789H2_CMD_PORCTRL, (uint8_t *) config->porch_param,
+			 sizeof(config->porch_param));
+
+	/* Digital Gamma Enable, default disabled */
+	tmp = 0x00;
+	st7789h2_transmit(dev, ST7789H2_CMD_DGMEN, &tmp, 1);
+
+	/* Frame Rate Control in Normal Mode, default value */
+	tmp = 0x1e; /* 40 Hz */
+	st7789h2_transmit(dev, ST7789H2_CMD_FRCTRL2, &tmp, 1);
+
+	st7789h2_transmit(dev, ST7789H2_CMD_GCTRL, (uint8_t *) config->gctrl, 1);
+
+	st7789h2_transmit(dev, ST7789H2_CMD_VCOMS, (uint8_t *) config->vcom, 1);
+
+	if (config->vdvvrhen) {
+		tmp = 0x01;
+		st7789h2_transmit(dev, ST7789H2_CMD_VDVVRHEN, &tmp, 1);
+		st7789h2_transmit(dev, ST7789H2_CMD_VRH, (uint8_t *) config->vrhs, 1);
+		st7789h2_transmit(dev, ST7789H2_CMD_VDS, (uint8_t *) config->vdvs, 1);
+	}
+
+	st7789h2_transmit(dev, ST7789H2_CMD_PWCTRL1, (uint8_t *) config->pwctrl1_param,
+			 sizeof(config->pwctrl1_param));
+
+	/* Memory Data Access Control */
+	st7789h2_transmit(dev, ST7789H2_CMD_MADCTL, (uint8_t *) config->mdac, 1);
+
+	/* Interface Pixel Format */
+	st7789h2_transmit(dev, ST7789H2_CMD_COLMOD, (uint8_t *) config->colmod, 1);
+
+	st7789h2_transmit(dev, ST7789H2_CMD_LCMCTRL, (uint8_t *) config->lcm, 1);
+
+	st7789h2_transmit(dev, ST7789H2_CMD_GAMSET, (uint8_t *) config->gamma, 1);
+
+	st7789h2_transmit(dev, ST7789H2_CMD_INV_ON, NULL, 0);
+
+	st7789h2_transmit(dev, ST7789H2_CMD_PVGAMCTRL, (uint8_t *) config->pvgam_param,
+			 sizeof(config->pvgam_param));
+
+	st7789h2_transmit(dev, ST7789H2_CMD_NVGAMCTRL, (uint8_t *) config->nvgam_param,
+			 sizeof(config->nvgam_param));
+
+	st7789h2_transmit(dev, ST7789H2_CMD_RAMCTRL, (uint8_t *) config->ram_param,
+			 sizeof(config->ram_param));
+
+	st7789h2_transmit(dev, ST7789H2_CMD_RGBCTRL, (uint8_t *) config->rgb_param,
+			 sizeof(config->rgb_param));
+}
+
+static int st7789h2_init(const struct device *dev)
+{
+	struct st7789h2_data *data = (struct st7789h2_data *)dev->data;
+	const struct st7789h2_config *config = (const struct st7789h2_config *)dev->config;
+	(void)data;
+	(void)config;
+
+	if (config->reset_gpio.port) {
```

Comment:
One thing is missing in this function: each GPIO port should be checked with `device_is_ready` before passing the DT spec to gpio_pin_configure_dt. If device_is_ready returns false for any required devices, you should fail with -ENODEV.

Otherwise my comments are all addressed, thank you for your patience.

Reply:
That has been fixed, now I check for device_is_ready before configuring :-)